application/service/base/base_service.go

```diff
@@ -9,6 +9,6 @@ type BaseService struct {
 	context.ServiceContext
 }
 
-func NewBaseService(context *context.ServiceContext) BaseService {
-	return BaseService{*context}
+func NewBaseService(context context.ServiceContext) BaseService {
```

Comment:
I'm concerned about this change - if we are passing the service context by value, not by pointer, then won't the state of that context be a copy of the service context parameter's state?  If that is so, then member values such as `transactionManager`, `inTransaction` etc will be different instances to those of the original parameter, meaning that an "inner" service will have a different context to an "outer" service, therefore won't be _aware_ of when the other service has started a transaction?  This would mean that all transactional code in services will be in an isolated transaction, right?

My understanding is based on this:

[http://goinbigdata.com/golang-pass-by-pointer-vs-pass-by-value/](http://goinbigdata.com/golang-pass-by-pointer-vs-pass-by-value/)

Reply:
As the result it was impossible to create a service passing nil as context.ServiceContext which was needed for tests.